src/strategies/erc721-with-tokenid-range-weights-simple/index.ts

```diff
@@ -19,48 +38,66 @@ export async function strategy(
   snapshot
 ) {
   const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const { tokenIdWeightRanges, defaultWeight } = options;
+  let erc721WeightedBalance = {};
+
+  const getDataFromBlockChain = async (
+    contractCalls: [string, string, [number]][]
+  ) => multicall(network, provider, abi, contractCalls, { blockTag });
+
+  const filterUnusedAddresses = (addressesToFilter: string[]): string[] =>
+    addressesToFilter.filter((address: string) =>
+      addresses
+        .map((address: string) => address.toLowerCase())
+        .includes(address.toLowerCase())
+    );
+
+  const multiplyOccurrencesByWeights = async (
+    contractCallResponse: [string, { owner: string }],
+    weights: number[]
+  ) =>
+    contractCallResponse
+      .map((address, index) => Array(weights[index]).fill(address[0]))
+      .flat();
+
+  const countOccurrences = (array: string[]) =>
+    array.reduce(
+      (prev, curr) => (prev[curr] ? ++prev[curr] : (prev[curr] = 1), prev),
+      {}
+    );
+
+  const getCustomRangeBalance = async (): Promise<{
+    [address: string]: number;
+  }> => {
+    const { ids, weights } = flattenTokenIdWeightRanges(tokenIdWeightRanges);
+    const contractCalls = ids.map((id: number) => [
+      options.address,
+      'ownerOf',
+      [id]
+    ]) as [string, string, [number]][];
+    const customRangeResponse = await getDataFromBlockChain(contractCalls);
```

Comment:
I am worried if this can create some problems while passing all the ids at once because previously this strategy was calling one range at once

Reply:
maybe we can create a new strategy? with some other easy name like `erc721-tokenid-ranges-with-weights`

Reply:
Yeah, this was on purpose. It's doing the same thing, just with fewer external calls and less time-consuming.

Before, the strategy did one external call for each range (limiting it to 5 ranges because Snapshot only allows 5 external calls) and only applied the wights in the final step.
Now, the strategy concatenates (flattened) the ranges in one single vector and creates a matching weights vector with `flattenTokenIdWeightRanges` to remove this limitation. It applies the weights to each range with a pairwise multiplication of the vectors in `multiplyOccurrencesByWeights`, before all other steps.

There was also one filter operation for every address and range, which led to an O(n^2) time complexity, limiting the maximum range to 7000. 
Now, the IDs are filtered once, and then the strategy counts the occurrences of each address, bringing the time complexity to O(n), improving performance.

Reply:
> maybe we can create a new strategy? with some other easy name like `erc721-tokenid-ranges-with-weights`

I can, but I've tried to not do that to mitigate clutter. It's basically the same strategy, it should be fully compliant, without the previous limitations.

Maybe @gregegan can give it a review to see if I've missed something.